docs/rendering_templates.md

```diff
@@ -0,0 +1,118 @@
+# Rendering BOSH Templates
+
+You can read more about BOSH templates on [bosh.io](https://bosh.io/docs/jobs/#templates).
+
+Rendering happens using `ExtendedJobs`.
+
+An `ExtendedJob` is required for each BOSH Release and Instance Group defined in a Desired Manifest.
+
+Details are omitted in the example below, for brevity:
+
+```yaml
+---
+
+releases:
+- name: release-a
+- name: release-b
+
+instance_groups:
+- name: instance-group-1
+  jobs:
+  - name: job1
+    release: release-a
+  - name: job2
+    release: release-a
+- name: instance-group-2
+  jobs:
+  - name: job1
+    release: release-a
+    properties:
+      foo: "((bar))"
+  - name: job3
+    release: release-b
+
+variables:
+- name: bar
+  type: password
+```
+
+In this case, to render all required templates, 3 `ExtendedJobs` are created:
+- one for  `instance-group-1`, because it only contains jobs from `release-a`
+- two for `instance-group-2`, because it references jobs from both `release-a` and  `release-b`
+
+The `ExtendedJob` runs a ruby process `TODO: reference to the implementation`
+
+## Input & Output
+
+Input is the desired manifest, generated `Secrets` for all referenced variables and information required for discovery of network addresses.
+
+We take advantage of the `ExtendedJob`'s feature to persist output in a `ConfigMap` or `Secret`.
+
+The output of the rendering process is a JSON object that contains all the rendered templates:
+
+The following is and example definition for the `ExtendedJob` that renders templates for `instance-group-2` and `release-a`.
+The release is inferred from the image itself.
```

Comment:
This currently doesn't work; requires https://www.pivotaltracker.com/story/show/163013866 to be finished